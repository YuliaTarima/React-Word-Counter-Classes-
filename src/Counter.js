/***
 * Excerpted from "React for Real",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material,
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose.
 * Visit http://www.pragmaticprogrammer.com/titles/lfreact for more book information.
***/
import React from 'react';

/*
function Counter({ count }) {
  return <p className="mb2">Word count: {count}</p>;
}
*/

function Counter({ count }) {
  return (
    <p className="mb2">
      <label htmlFor="count">Word count: </label>
      <output id="count">
        {count}
      </output>
    </p>
  );
}

export default Counter;
