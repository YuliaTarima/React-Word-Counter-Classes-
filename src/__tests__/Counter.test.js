/***
 * Excerpted from "React for Real",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material,
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose.
 * Visit http://www.pragmaticprogrammer.com/titles/lfreact for more book information.
***/
import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../Counter';

describe('A counter', () => {
  it('Displays the count and label', () => {
    const counter = renderer.create(<Counter legend="Count" count={34} />);
    expect(counter.toJSON()).toMatchSnapshot();

  });
});
