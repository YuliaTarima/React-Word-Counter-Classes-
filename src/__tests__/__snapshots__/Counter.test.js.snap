// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A counter Displays the count and label 1`] = `
<p
  className="mb2"
>
  <label
    htmlFor="count"
  >
    Word count: 
  </label>
  <output
    id="count"
  >
    34
  </output>
</p>
`;
